### PR TITLE
add more info to fragements and extension

### PR DESCRIPTION
**Pull request by aevesdocker**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

**Review by milas — APPROVED** [review]

File: 10-fragments.md [hunk 1]
```diff
@@ -33,3 +63,22 @@ volumes:
...
+      - CONFIG_KEY
+      - EXAMPLE_KEY
+      - DEMO_VAR
+  second:
+    image: another-image:latest
+    environment:
+      <<: *env
+      - AN_EXTRA_KEY
+      - SECOND_SPECIFIC_KEY
+```
```
Comment (ndeloof) [inline 10-fragments.md:82]: this doesn't work. yaml merge `<<: *xx` can only be used with a mapping.

**Review by ndeloof — CHANGES_REQUESTED** [review]
way better than previous doc
but example merging with a sequence is invalid

**Comment by aevesdocker** [comment]
Thanks for catching that @ndeloof. I've updated the example with bits from your closed PR

**Comment by aevesdocker** [comment]
Unsure why I'm suddenly getting build errors again and spec.md file appears deleted 🤔

**Review by ndeloof — APPROVED** [review]

**Pull request merged**